sifive-blocks/templates/metal/platform/metal_platform_sifive_hca0.h.j2

```diff
@@ -0,0 +1,30 @@
+/* Copyright 2020 SiFive, Inc */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#ifndef METAL__PLATFORM__SIFIVE_HCA0_H
+#define METAL__PLATFORM__SIFIVE_HCA0_H
+
+{% if 'sifive,hca0' in devices %}
+{% set sifive_hca0s = devices['sifive,hca0'] %}
+
+{% for hca in sifive_hca0s %}
+    {% if hca.reg_names is defined %}
+#define METAL_SIFIVE_HCA0_{{ loop.index0 }}_BASE_ADDRESS {{ '0x%x' % hca.regs_by_name["control"][0] }}
+#define METAL_SIFIVE_HCA0_{{ loop.index0 }}_SIZE {{ '0x%x' % hca.regs_by_name["control"][1] }}
+    {% else %}
+#define METAL_SIFIVE_HCA0_{{ loop.index0 }}_BASE_ADDRESS {{ '0x%x' % hca.reg[0][0] }}
+#define METAL_SIFIVE_HCA0_{{ loop.index0 }}_SIZE {{ '0x%x' % hca.reg[0][1] }}
+    {% endif %}
+{% endfor %}
+
+#define METAL_SIFIVE_HCA0
+
+typedef enum
+{
+    HCA_AES_MODE = 0,
+    HCA_SHA_MODE = 1
+} hca0_mode_t;
```

Comment:
I don't see where this enum type is used, do we need it? If we do, maybe it should live in a static header file instead of a generated one, since I also don't think it's part of the "platform" header content.

Reply:
It is used into sifive_hca0_aes_cipher, sifive_hca0_aes_auth_init, sifive_hca0_sha_init even if in functio sifive_hca0_aes_cipher, sifive_hca0_aes_auth_init it is commented (because set to 0 the field when you just reset it just before is not needed)